test/common/testlib.py

```diff
@@ -466,9 +466,13 @@ def run(self, result=None):
                 self.currentResult = result
                 super(MachineCase, self).run(result)
             except RetryError, ex:
-                self.doCleanups()
-                sys.stderr.write("{0}\n".format(ex))
-                continue
+                if retry < 5:
```

Comment:
Yes, but if we want to enforce a upper limit as well, we should fail when the upper limit is reached, no?